capabilityreceiver/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -262,6 +267,22 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 	return nil
 }
 
+func removeNoLongerExistingVersionsInfos(ctx context.Context, store *postgresql.Store, url string, supportedVersions []string) error {
+	// If there is a requestedVersion for a URL in fhir_endpoints_info that is no longer in supportedVersions
+	// then we need to remove those fhir_endpoint_info entries
+	endptInfos, err := store.GetFHIREndpointInfosByURLWithDifferentRequestedVersion(ctx, url, supportedVersions)
+	if err != nil {
+		return err
+	}
+	for _, infoEntry := range endptInfos {
+		err = store.DeleteFHIREndpointInfo(ctx, infoEntry)
```

Comment:
This is just a question and I think is kind of out of scope of this ticket, but why does `DeleteFHIREndpointInfo` take a whole FhirEndpointInfo object? It seems like kind of overkill to have to get all of the fields of the table in `GetFHIREndpointInfosByURLWithDifferentRequestedVersion` when we really just need the IDs? I was just wondering why `DeleteFHIREndpointInfo` was set up that way and if there's something I'm missing

Reply:
Good question, I don't see why there couldn't be a "DeleteFHIREndpoitInfoByID" -- But I think since everywhere we use the DeleteFHIREndpointInfo we have the entire entry the decision of whether to take in the whole entry or just the ID is 6 of one, half dozen of the other

Reply:
yeah that's fair, i was just curious!